components/browser/menu/src/main/java/mozilla/components/browser/menu/view/MenuButton.kt

```diff
@@ -0,0 +1,141 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.menu.view
+
+import android.content.Context
+import android.content.res.ColorStateList
+import android.util.AttributeSet
+import android.view.View
+import android.widget.FrameLayout
+import android.widget.ImageView
+import androidx.annotation.ColorInt
+import androidx.annotation.VisibleForTesting
+import androidx.core.content.ContextCompat
+import mozilla.components.browser.menu.BrowserMenu
+import mozilla.components.browser.menu.BrowserMenu.Orientation
+import mozilla.components.browser.menu.BrowserMenuBuilder
+import mozilla.components.browser.menu.BrowserMenuHighlight
+import mozilla.components.browser.menu.R
+
+/**
+ * A `three-dot` button used for expanding menus.
+ *
+ * If you are using a browser toolbar, do not use this class directly.
```

Comment:
Nit: up to you, but I would consider renaming this to `MenuButtonImpl`, just to further reinforce the point that this really isn't intended for public consumption.  Might also add something like this to the kdoc:

-----
WARNING: Prefer using a browser toolbar, which internally delegates to this class.

If you need this class (i.e., browser toolbar does not support one of your requirements), you will have to trigger most of its behavior manually, including invalidating menu items, setting highlighted state, etc.  See `browser-toolbar` `MenuButton` for example usage.

Reply:
Part of the reason I rewrote this component was to make it more suited to public consumption :) I personally prefer to leave the current name, as I don't think `Impl` is a good name for a view.

The invalidation work is intentionally pulled out because it's not needed for the button itself, and I want to make it more generic so that its easier to integrate with the new menu component I've written later.